### ISSUE TITLE
SecretKey parsing

### ISSUE THREAD

**Issue by peterromfeldhk** [funcool/buddy-core#15]
i recently tried to use buddy.sign together with buddy.core.keys to generate a JWT for google service account.

google service account you just get a JSON with a privatekey as string, passing it to buddy.core.keys/str->privatekey didnt work. using another lib did the job.

check out https://github.com/xsc/pem-reader for enhancing buddy.core.keys ;)

**Comment by niwinz** [comment]
Hi @peterromfeldhk 

Can you provide a failing test case for that? Buddy uses bouncy castle for read pems, and pem-reader uses only jdk, si I find it a little bit strange that you are saying.

Thanks.

**Comment by peterromfeldhk** [comment]
Hi @niwinz 

Im on it, 
fyi:
```
(require '[pem-reader.core :as pem])
(require '[buddy.core.keys :as keys])

(-> (slurp "/Users/peter/Repositories/private/buddy-core/test/_files/privkey.pkcs8.pem")
    .getBytes
    byte-array
    io/input-stream
    pem/read
    pem/private-key
    keys/private-key?)
=> true
(-> (slurp "/Users/peter/Repositories/private/buddy-core/test/_files/privkey.pkcs8.pem")
    .getBytes
    byte-array
    io/input-stream
    pem/read
    pem/private-key
    type)
=> sun.security.rsa.RSAPrivateCrtKeyImpl
(-> (slurp "/Users/peter/Repositories/private/buddy-core/test/_files/privkey.pkcs8.pem")
    keys/str->private-key)
java.lang.ClassCastException: org.bouncycastle.asn1.pkcs.PrivateKeyInfo cannot be cast to org.bouncycastle.openssl.PEMKeyPair
```

**Comment by niwinz** [comment]
Thanks!

**Comment by peterromfeldhk** [comment]
couldnt endure failing test lol:
```
(let [converter (doto (JcaPEMKeyConverter.)
                  (.setProvider "BC"))]
  (->> (slurp "/Users/peter/Repositories/private/buddy-core/test/_files/privkey.pkcs8.pem")
       StringReader.
       io/reader
       PEMParser.
       .readObject
       (.getPrivateKey converter)
       keys/private-key?))
=> true
```

**Comment by niwinz** [comment]
Should be fixed with #20
